src/logging.cpp

```diff
@@ -97,22 +103,67 @@ namespace logging {
     os << "["sv << std::put_time(&lt, "%Y-%m-%d %H:%M:%S.") << boost::format("%03u") % ms.count() << "]: "sv
        << log_type << view.attribute_values()[message].extract<std::string>();
   }
+#ifdef __ANDROID__
+  namespace sinks = boost::log::sinks;
+  namespace expr = boost::log::expressions;
+
+  void android_log(const std::string &message, int severity) {
+    android_LogPriority android_priority;
+    switch (severity) {
+      case 0:
+        android_priority = ANDROID_LOG_VERBOSE;
+        break;
+      case 1:
+        android_priority = ANDROID_LOG_DEBUG;
+        break;
+      case 2:
+        android_priority = ANDROID_LOG_INFO;
+        break;
+      case 3:
+        android_priority = ANDROID_LOG_WARN;
+        break;
+      case 4:
+        android_priority = ANDROID_LOG_ERROR;
+        break;
+      case 5:
+        android_priority = ANDROID_LOG_FATAL;
+        break;
+      default:
+        android_priority = ANDROID_LOG_UNKNOWN;
+        break;
+    }
+    __android_log_print(android_priority, "Sunshine", "%s", message.c_str());
+  }
+
+  // custom sink backend for android
+  struct android_sink_backend: public sinks::basic_sink_backend<sinks::concurrent_feeding> {
+    void consume(const bl::record_view &rec) {
+      int log_sev = rec[severity].get();
+      const std::string log_msg = rec[expr::smessage].get();
+      // log to android
+      android_log(log_msg, log_sev);
+    }
+  };
+#endif
 
   [[nodiscard]] std::unique_ptr<deinit_t> init(int min_log_level, const std::string &log_file) {
     if (sink) {
       // Deinitialize the logging system before reinitializing it. This can probably only ever be hit in tests.
       deinit();
     }
 
+#ifndef __ANDROID__
     setup_av_logging(min_log_level);
     setup_libdisplaydevice_logging(min_log_level);
+#endif
```

Comment:
Does it compile if you remove the `ifndef`?

Reply:
It seems it's possible to use FFmpeg on Android with hw encoding. I didn't do a ton of research on this, but a quick search brought this up. https://www.reddit.com/r/androiddev/comments/clz0ol/comment/evz6vlq/

I think we can probably add an Android build off FFmpeg to our pre-built FFmpeg (https://github.com/LizardByte/build-deps). We're already doing some cross-compiling there, though I have not looked into doing this for android yet. This can definitely come later if you have a different approach for now.

Reply:
ffmpeg would be really nice for us at blissOS as we support amd and intel gpus. Even for cpu encoding ffmpeg would be nice instead of relying on platform support.

I don't have a whole bunch of time to help out with this, but media-kit and mpv-android cross compile ffmpeg. You can find the buildscripts for mpv-android here https://github.com/mpv-android/mpv-android/tree/master/buildscripts.